docs/system/bms.md

```diff
@@ -1,8 +1,62 @@
+
+## Main Functions
+
+### Electric Cell Protection
+
+As described in chapter ["Battery"](battery.md), different cells have different minimum and maximum voltage levels as well as different phases while charging/discharging.
+The BMS is responsible to measure the voltage, current and temperature and calculate the state of charge (SOC) of the cells are in order to apply the correct current or voltage.
+It will optimise the charging rate and determine when to stop charging/discharging.
+
+To disconnect the battery from the load/charger, different types of switches can be used. For low current systems, a MOSFET switch is often easiest to use, while a mechanical or solid state relais can be necessary to switch higher voltages and currents.
+
+
+
+### Thermal Protection
+
+In order to keep the battery in a safe operational state, the thermal managmenent is vital. The system can either be active or passive, using mostly air or another liquid coolant. The necessary hardware like fans and their electronics are mostly not considered part of the BMS but are controlled by it.
+
+
+### Balancing
+
+To optimize performance and lifetime of battery packs, balancing is used to distribute load and prevent localized over- or under-charging of cells. Since cells differ slightly in capacity depending on the quality of the manufacturing process and used materials or by cell-aging, cells connected in series can show different voltages. To normalize the SOC for each cell, a BMS can either actively transfer energy from higher charged cells to those with lower SOC or passivly, mostly by wasting energy from cells with 100% SOC until every cell is fully charged.
+
+Passive balancing can be done by bleeding energy to ground through a resistor and a transistor or with a resistor parallel to the cell which acts like a small consumer. Passive balancing increases the thermal energy dissipated by the pack and decreases the overall efficieny.
+
+Active balancing aims to distribute the energy better while charging and discharging, but need much more circuitry.
+
+### Communication
+
+A BMS can also feature communication capabilities to distribute information to other BMS or [charge-controllers](charge_controller.md) or logging systems. Commonly used for standalone setups is a serial connection while BUS connections like CAN are preferably used in systems with several components. See chapter [Communication](../development/communication.md) for more.
+
+### Bus precharge
+
+A common feature may also be the connection to a load, including a precharge system. Precharging protects the battery against high inrush currents when a load is connected. To slowly charge large capacities in the load circuit, the load is connected over a resistor for a short period of time. Afterwards, the resistor is removed and the load connected normally. 
```

Comment:
The inrush current goes into the capacitors, not the battery. From battery perspective it would be outrush currents, but that term is not used. And it protects the capacitors, not the battery.